src/dd/dd.rs

```diff
@@ -0,0 +1,734 @@
+             cbs: {cbs}\n\
+             skip: {skip}\n\
+             seek: {seek}\n\
+             count: {count}\n\
+             flags: {flags}\n",
+            input_filename = self.inp_name,
+            output_filename = self.out_name,
+            ibs = self.inp_block_size,
+            obs = self.out_block_size,
+            cbs = self.cnv_block_size,
+            skip = self.skip,
+            seek = self.seek,
+            count = self.count,
+            flags = self.str_flags()
+        )
+    }
+}
+
+pub fn parse_args() -> Conf {
+    let mut cf = Conf::default();
+    for arg in env::args().skip(1) {
+        let pair: Vec<&str> = arg.splitn(2, '=').collect();
+        if pair.len() == 2 {
+            let (key, val) = (pair[0], pair[1]);
+            match key {
+                "if" => cf.set_if(val),
+                "of" => cf.set_of(val),
+                "conv" => for elem in val.split(",") {
+                    match Conf::get_flag(elem) {
+                        Conf::INVALID_FLAG => err("invalid flag passed to conv"),
+                        flag => cf.set_flag(flag),
+                    }
+                },
+                _ => {
+                    if Conf::is_flag_valid(key) {
+                        if let Some(m) = get_u(val) {
+                            match key {
+                                "ibs" => cf.inp_block_size = m,
+                                "obs" => cf.out_block_size = m,
+                                "bs" => {
+                                    cf.inp_block_size = m;
+                                    cf.out_block_size = m;
+                                }
+                                "cbs" => cf.cnv_block_size = m,
+                                "skip" => cf.skip = m,
+                                "seek" => cf.seek = m,
+                                "count" => cf.count = m,
+                                _ => err(format!("invalid argument: {}", key)),
+                            }
+                        }
+                    } else {
+                        err(format!("invalid argument: {}", arg));
+                    }
+                }
+            }
+        } else {
+            err(format!("invalid argument: {}", arg));
+        }
+    }
+
+    // Behaviour is unspecified. We choose to terminate.
+    if cf.count == 0 {
+        err("assign count non-zero value");
+    }
+
+    if cf.is_stdout && cf.seek != 0 {
+        err("cannot seek on standard output");
+    }
+
+    if cf.has_flag(Conf::BLOCK) && cf.has_flag(Conf::UNBLOCK) {
+        err("'block' and 'unblock' values are mutually exclusive");
+    }
+
+    if cf.has_flag(Conf::LCASE) && cf.has_flag(Conf::UCASE) {
+        err("'lcase' and 'ucase' values are mutually exclusive");
+    }
+
+    if !(cf.flags & Conf::BLOCK != 0 || cf.flags & Conf::UNBLOCK != 0 || cf.flags & Conf::LCASE != 0
+        || cf.flags & Conf::UCASE != 0 || cf.flags & Conf::SWAB != 0)
+        {
```

Comment:
The braces should line up with the `if`.  Also the second line of conditions should probably be indented one more time.

Reply:
@Arcterus, _rustfmt_ leaves indentation of second line as is.
Should I indent it one more time?

Reply:
It’s fine as is.

Reply:
Nice, I left it indented once.